src/main/java/org/jfrog/hudson/pipeline/common/executors/XrayExecutor.java

```diff
@@ -0,0 +1,85 @@
+package org.jfrog.hudson.pipeline.common.executors;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import org.apache.commons.cli.MissingArgumentException;
+import org.apache.commons.lang.StringUtils;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.client.ProxyConfiguration;
+import org.jfrog.build.client.artifactoryXrayResponse.ArtifactoryXrayResponse;
+import org.jfrog.build.extractor.clientConfiguration.client.ArtifactoryXrayClient;
+import org.jfrog.hudson.CredentialsConfig;
+import org.jfrog.hudson.pipeline.common.Utils;
+import org.jfrog.hudson.pipeline.common.types.ArtifactoryServer;
+import org.jfrog.hudson.pipeline.common.types.XrayScanConfig;
+import org.jfrog.hudson.pipeline.common.types.XrayScanResult;
+import org.jfrog.hudson.util.JenkinsBuildInfoLog;
+
+/**
+ * @author Alexei Vainshtein
+ */
+public class XrayExecutor implements Executor {
+
+    private XrayScanConfig xrayScanConfig;
+    private TaskListener listener;
+    private ArtifactoryServer server;
+    private Run build;
+    private XrayScanResult xrayScanResult;
+
+    public XrayExecutor(XrayScanConfig xrayScanConfig, TaskListener listener, ArtifactoryServer server, Run build) {
+        this.xrayScanConfig = xrayScanConfig;
+        this.listener = listener;
+        this.server = server;
+        this.build = build;
+    }
+
+    @Override
+    public void execute() throws Exception {
+        if (StringUtils.isEmpty(xrayScanConfig.getBuildName())) {
+            throw new MissingArgumentException("Xray scan build name is mandatory");
+        }
+
+        if (StringUtils.isEmpty(xrayScanConfig.getBuildNumber())) {
```

Comment:
I think that these 2 checks are already made in `createXrayScanConfig():
```if (!xrayScanParams.keySet().containsAll(mandatoryArgumentsAsList)) {```

Reply:
Right I removed it from the executor